lib.sh

```diff
@@ -1159,6 +1147,18 @@ do
 done
 }
 
+# Helper function for generating random passwords
+gen_passwd() {
+    local length=$1
+    local charset="$2"
+    local password=""
+    while [ ${#password} -lt "$length" ]
+    do
+        password=$(echo "$password""$(head -c 100 /dev/urandom | LC_ALL=C tr -dc "$charset")" | fold -w "$length" | head -n 1)
+    done
+    print_text_in_color "$Cyan" "$password"
```

Comment:
When you `print_text_in_color`, doesn't it actually change the output of this function, which is used as input when evaluating global variables? So now the passwords instead of "abcdefg1234" will be "<cyan escape chars>abcdefg1234<reset color escape chars>"?

Reply:
In other words: the echo was there to "return" the generated password, I don't really want to print it on screen in this place ;)

Reply:
I tried to run it, following errors are printed right after beginning of the installation:
`/dev/fd/63: line 30: print_text_in_color: command not found`

Reply:
I reverted my changes as I realized it wouldn't work.